src/operator/controllers/intents_reconcilers/networkpolicy/builders/dns_egress_network_policy.go

```diff
@@ -0,0 +1,46 @@
+package builders
+
+import (
+	"context"
+	otterizev1alpha3 "github.com/otterize/intents-operator/src/operator/api/v1alpha3"
+	"github.com/otterize/intents-operator/src/operator/effectivepolicy"
+	"github.com/otterize/intents-operator/src/shared/injectablerecorder"
+	"github.com/samber/lo"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/networking/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+// The DNSEgressNetworkPolicyBuilder creates network policies that allow egress traffic from pods.
+type DNSEgressNetworkPolicyBuilder struct {
+	injectablerecorder.InjectableRecorder
+}
+
+func NewDNSEgressNetworkPolicyBuilder() *DNSEgressNetworkPolicyBuilder {
+	return &DNSEgressNetworkPolicyBuilder{}
+}
+
+// a function that creates []NetworkPolicyEgressRule from ep
+func (r *DNSEgressNetworkPolicyBuilder) buildNetworkPolicyEgressRules(ep effectivepolicy.ServiceEffectivePolicy) []v1.NetworkPolicyEgressRule {
+	if lo.NoneBy(ep.Calls, func(call otterizev1alpha3.Intent) bool {
+		return call.Type == "" || call.Type == otterizev1alpha3.IntentTypeHTTP || call.Type == otterizev1alpha3.IntentTypeKafka
```

Comment:
you can use call.IsTargetInCluster()
```suggestion
		return call.IsTargetInCluster()
```